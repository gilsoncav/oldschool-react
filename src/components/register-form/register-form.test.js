import RegisterForm from './register-form';

jest.mock('../../service/auth-service.js', () => ({
  ...jest.requireActual('../../service/auth-service.js'),
  registerUser: jest.fn().mockReturnValueOnce(true),
}));

describe('RegisterForm component', () => {
  const spyHistPush = sinon.spy();
  const historyMock = { push: spyHistPush };

  it('renders correctly', () => {
    const testForm = {
      getFieldValue: jest.fn(opts => c => c),
      getFieldDecorator: jest.fn(opts => c => c),
      validateFields: jest.fn(),
    };
    const wrapper = shallow(
      <RegisterForm.WrappedComponent history={historyMock} form={testForm} />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should handleSubmit and simulate success on register, redirect to dashboard', async (done) => {
    const testForm = {
      getFieldValue: jest.fn(opts => c => c),
      getFieldDecorator: jest.fn(opts => jest.fn()),
      validateFields: jest.fn(callback => callback(false, [])),
    };

    const wrapper = shallow(
      <RegisterForm.WrappedComponent history={historyMock} form={testForm} />,
    );
    await wrapper.instance().handleSubmit(new Event('submit'));
    expect(spyHistPush.calledOnce).toBe(true);
    expect(spyHistPush.args[0][0]).toBe('/dashboard');
    done();
  });
});
