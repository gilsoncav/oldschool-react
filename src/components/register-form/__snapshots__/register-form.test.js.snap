// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm component renders correctly 1`] = `
.c0 > #register-form #submit-register {
  width: 100%;
  margin-top: 15px;
}

<div
  className="c0"
>
  <form
    className="ant-form ant-form-horizontal"
    id="register-form"
    onSubmit={[Function]}
  >
    <div
      className="ant-row ant-form-item"
    >
      <div
        className="ant-col ant-col-20 ant-col-offset-2 ant-form-item-label ant-form-item-label-left"
      >
        <label
          className=""
          htmlFor={null}
          onClick={[Function]}
          title="Name"
        >
          Name
        </label>
      </div>
      <div
        className="ant-col ant-col-20 ant-col-offset-2 ant-form-item-control-wrapper"
      >
        <div
          className="ant-form-item-control"
        >
          <span
            className="ant-form-item-children"
          >
            <span
              className="ant-input-affix-wrapper"
            >
              <span
                className="ant-input-prefix"
              >
                <i
                  aria-label="icon: user"
                  className="anticon anticon-user"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="user"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M858.5 763.6a374 374 0 0 0-80.6-119.5 375.63 375.63 0 0 0-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 0 0-80.6 119.5A371.7 371.7 0 0 0 136 901.8a8 8 0 0 0 8 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 0 0 8-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                    />
                  </svg>
                </i>
              </span>
              <input
                className="ant-input"
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder="Name"
                style={null}
                type="text"
                value=""
              />
            </span>
          </span>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
    >
      <div
        className="ant-col ant-col-20 ant-col-offset-2 ant-form-item-label ant-form-item-label-left"
      >
        <label
          className=""
          htmlFor={null}
          onClick={[Function]}
          title="Username"
        >
          Username
        </label>
      </div>
      <div
        className="ant-col ant-col-20 ant-col-offset-2 ant-form-item-control-wrapper"
      >
        <div
          className="ant-form-item-control"
        >
          <span
            className="ant-form-item-children"
          >
            <span
              className="ant-input-affix-wrapper"
            >
              <span
                className="ant-input-prefix"
              >
                <i
                  aria-label="icon: user"
                  className="anticon anticon-user"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="user"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M858.5 763.6a374 374 0 0 0-80.6-119.5 375.63 375.63 0 0 0-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 0 0-80.6 119.5A371.7 371.7 0 0 0 136 901.8a8 8 0 0 0 8 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 0 0 8-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                    />
                  </svg>
                </i>
              </span>
              <input
                autoComplete="username"
                className="ant-input"
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder="Username"
                style={null}
                type="text"
                value=""
              />
            </span>
          </span>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
    >
      <div
        className="ant-col ant-col-20 ant-col-offset-2 ant-form-item-label ant-form-item-label-left"
      >
        <label
          className=""
          htmlFor={null}
          onClick={[Function]}
          title="E-mail"
        >
          E-mail
        </label>
      </div>
      <div
        className="ant-col ant-col-20 ant-col-offset-2 ant-form-item-control-wrapper"
      >
        <div
          className="ant-form-item-control"
        >
          <span
            className="ant-form-item-children"
          >
            <span
              className="ant-input-affix-wrapper"
            >
              <span
                className="ant-input-prefix"
              >
                <i
                  aria-label="icon: mail"
                  className="anticon anticon-mail"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="mail"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M928 160H96c-17.7 0-32 14.3-32 32v640c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V192c0-17.7-14.3-32-32-32zm-40 110.8V792H136V270.8l-27.6-21.5 39.3-50.5 42.8 33.3h643.1l42.8-33.3 39.3 50.5-27.7 21.5zM833.6 232L512 482 190.4 232l-42.8-33.3-39.3 50.5 27.6 21.5 341.6 265.6a55.99 55.99 0 0 0 68.7 0L888 270.8l27.6-21.5-39.3-50.5-42.7 33.2z"
                    />
                  </svg>
                </i>
              </span>
              <input
                className="ant-input"
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder="E-mail"
                style={null}
                type="text"
                value=""
              />
            </span>
          </span>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
    >
      <div
        className="ant-col ant-col-20 ant-col-offset-2 ant-form-item-label ant-form-item-label-left"
      >
        <label
          className=""
          htmlFor={null}
          onClick={[Function]}
          title="Password"
        >
          Password
        </label>
      </div>
      <div
        className="ant-col ant-col-20 ant-col-offset-2 ant-form-item-control-wrapper"
      >
        <div
          className="ant-form-item-control"
        >
          <span
            className="ant-form-item-children"
          >
            <span
              className="ant-input-password ant-input-affix-wrapper"
            >
              <span
                className="ant-input-prefix"
              >
                <i
                  aria-label="icon: lock"
                  className="anticon anticon-lock"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="lock"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32zM332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332V240zm460 600H232V536h560v304zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 1 0-56 0z"
                    />
                  </svg>
                </i>
              </span>
              <input
                action="click"
                autoComplete="new-password"
                className="ant-input"
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder="Password"
                style={null}
                type="password"
                value=""
              />
              <span
                className="ant-input-suffix"
              >
                <i
                  aria-label="icon: eye-invisible"
                  className="anticon anticon-eye-invisible ant-input-password-icon"
                  onClick={[Function]}
                  onMouseDown={[Function]}
                  tabIndex={-1}
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="eye-invisible"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 0 0 0-51.5zm-63.57-320.64L836 122.88a8 8 0 0 0-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 0 0 0 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 0 0 0 11.31L155.17 889a8 8 0 0 0 11.31 0l712.15-712.12a8 8 0 0 0 0-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 0 0-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 0 1 146.2-106.69L401.31 546.2A112 112 0 0 1 396 512z"
                    />
                    <path
                      d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 0 0 227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 0 1-112 112z"
                    />
                  </svg>
                </i>
              </span>
            </span>
          </span>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
    >
      <div
        className="ant-col ant-col-20 ant-col-offset-2 ant-form-item-label ant-form-item-label-left"
      >
        <label
          className=""
          htmlFor={null}
          onClick={[Function]}
          title="Confirm Password"
        >
          Confirm Password
        </label>
      </div>
      <div
        className="ant-col ant-col-20 ant-col-offset-2 ant-form-item-control-wrapper"
      >
        <div
          className="ant-form-item-control"
        >
          <span
            className="ant-form-item-children"
          >
            <span
              className="ant-input-password ant-input-affix-wrapper"
            >
              <span
                className="ant-input-prefix"
              >
                <i
                  aria-label="icon: lock"
                  className="anticon anticon-lock"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="lock"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32zM332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332V240zm460 600H232V536h560v304zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 1 0-56 0z"
                    />
                  </svg>
                </i>
              </span>
              <input
                action="click"
                autoComplete="new-password"
                className="ant-input"
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder="Confirm Password"
                style={null}
                type="password"
                value=""
              />
              <span
                className="ant-input-suffix"
              >
                <i
                  aria-label="icon: eye-invisible"
                  className="anticon anticon-eye-invisible ant-input-password-icon"
                  onClick={[Function]}
                  onMouseDown={[Function]}
                  tabIndex={-1}
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="eye-invisible"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 0 0 0-51.5zm-63.57-320.64L836 122.88a8 8 0 0 0-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 0 0 0 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 0 0 0 11.31L155.17 889a8 8 0 0 0 11.31 0l712.15-712.12a8 8 0 0 0 0-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 0 0-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 0 1 146.2-106.69L401.31 546.2A112 112 0 0 1 396 512z"
                    />
                    <path
                      d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 0 0 227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 0 1-112 112z"
                    />
                  </svg>
                </i>
              </span>
            </span>
          </span>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
    >
      <div
        className="ant-col ant-col-20 ant-col-offset-2 ant-form-item-control-wrapper"
      >
        <div
          className="ant-form-item-control"
        >
          <span
            className="ant-form-item-children"
          >
            <button
              className="ant-btn ant-btn-primary"
              data-loading="loading"
              id="submit-register"
              onClick={[Function]}
              type="submit"
            >
              <span>
                Register
              </span>
            </button>
          </span>
        </div>
      </div>
    </div>
  </form>
</div>
`;
